l10n_it_asset_history_import/wizards/asset_history_import.py

```diff
@@ -434,15 +454,29 @@ def convert_to_asset_vals(self, asset_dict, workbook, sheet):
                     "Is it in its ISO 4217 format?"
                 )
             )
+        supplier_vat_col = get_header_by_model_and_field("res.partner", "vat").col
+        supplier_vat = asset_dict[supplier_vat_col]
+        partner = self.env["res.partner"]
+        if supplier_vat:
+            partner = partner.search([("vat", "ilike", supplier_vat)], limit=1)
+            if not partner:
+                raise ValidationError(
+                    _("Could not find Supplier with VAT %s", supplier_vat)
+                )
 
+        supplier_doc_ref_col = get_header_by_model_and_field(
+            "asset.asset", "supplier_ref"
+        ).col
+        supplier_doc_ref = asset_dict[supplier_doc_ref_col]
         vals.update(
             {
                 "category_id": categ.id,
                 "company_id": self.company_id.id,
                 "currency_id": curr.id,
+                "supplier_id": partner and partner.id or False,
```

Comment:
Don't need to check if the partner has been found:
1. since we defined `partner = self.env["res.partner"]`, then `partner.id` will be `False` anyway if no partner was found
2. also, raising the `ValidationError` if no partner is found, makes sure that we reach this part of the code only if `partner` is correctly valued
```suggestion
                "supplier_id": partner.id,
```